docs/content/reference/routing-configuration/http/middlewares/headers.md

```diff
@@ -0,0 +1,299 @@
+  - "traefik.http.middlewares.testheader.headers.accesscontrolalloworiginlist=https://foo.bar.org,https://example.org"
+  - "traefik.http.middlewares.testheader.headers.accesscontrolmaxage=100"
+  - "traefik.http.middlewares.testheader.headers.addvaryheader=true"
+```
+
+```yaml tab="Consul Catalog"
+- "traefik.http.middlewares.testheader.headers.accesscontrolallowmethods=GET,OPTIONS,PUT"
+- "traefik.http.middlewares.testheader.headers.accesscontrolallowheaders=*"
+- "traefik.http.middlewares.testheader.headers.accesscontrolalloworiginlist=https://foo.bar.org,https://example.org"
+- "traefik.http.middlewares.testheader.headers.accesscontrolmaxage=100"
+- "traefik.http.middlewares.testheader.headers.addvaryheader=true"
+```
+
+## Configuration Options
+
+!!! warning
+
+    Custom headers will overwrite existing headers if they have identical names.
+
+!!! note ""
+
+    The detailed documentation for security headers can be found in [unrolled/secure](https://github.com/unrolled/secure#available-options).
+
+Field                         | Description                                       | Default   | Required
+----------------------------- | ------------------------------------------------- | --------- | --------
+`customRequestHeaders`          | Lists the header names and values for requests.  | | No
+`customResponseHeaders`         | Lists the header names and values for responses. |  | No
+`accessControlAllowCredentials` | Indicates if the request can include user credentials.| false     | No
+`accessControlAllowHeaders`     | Specifies allowed request header names.          | | No
+`accessControlAllowMethods`     | Specifies allowed request methods.               | | No
+`accessControlAllowOriginList`  | Specifies allowed origins. More information [here](#accesscontrolalloworiginlist)      | None      | No
+`accessControlAllowOriginListRegex` | Allows origins matching regex. More information [here](#accesscontrolalloworiginlistregex)            | None      | No
+`accessControlExposeHeaders`    | Specifies which headers are safe to expose to the API of a CORS API specification.       |      | No
+`accessControlMaxAge`           | Time (in seconds) to cache preflight requests.   | 0         | No
+`addVaryHeader`                | used in conjunction with `accessControlAllowOriginList` to determine whether the `Vary` header should be added or modified to demonstrate that server responses can differ based on the value of the origin header. | false     | No
+`allowedHosts`                  | Lists allowed domain names.                      | None      | No
+`hostsProxyHeaders`             | Specifies header keys for proxied hostname.      | None      | No
+`sslProxyHeaders`               | Defines a set of header keys with associated values that would indicate a valid HTTPS request. It can be useful when using other proxies (example: `"X-Forwarded-Proto": "https"`).        |      | No
```

Comment:
```suggestion
`sslProxyHeaders`               | Defines a set of header keys with associated values that would indicate a valid HTTPS request. It can be useful when using other proxies (example: `"X-Forwarded-Proto": "https"`).        | {}     | No
```